src/helpers/formatters/formatFootprint.ts

```diff
@@ -0,0 +1,58 @@
+import { Metric } from '@/publicodes-state/types'
+import { formatCarbonFootprint } from './formatCarbonFootprint'
+import { formatWaterFootprint } from './formatWaterFootprint'
+
+type Options = {
+  localize?: boolean
+  locale?: string
+  maximumFractionDigits?: number
+  shouldUseAbbreviation?: boolean
+  t?: (key: string) => string
+  metric?: Metric
+  shouldDivideBy365?: boolean
+}
+
+export function formatFootprint(
+  value: string | number,
+  {
+    localize,
+    locale,
+    maximumFractionDigits,
+    shouldUseAbbreviation,
+    t,
+    metric,
+    shouldDivideBy365,
+  }: Options = {}
+): {
+  formattedValue: string
+  unit: string | null
+  negative: boolean
+} {
+  if (metric === 'carbone') {
```

Comment:
switch case ? With that you can default to `carbon`

Reply:
Je déteste les switch (mais tu as raison)

Reply:
Fais une map sinon